examples/host/main.go

```diff
@@ -3,23 +3,75 @@ package main
 import (
 	"log"
 
+	c "github.com/aerogear/charmil/core/config"
+	"github.com/aerogear/charmil/examples/plugins/adder"
 	"github.com/aerogear/charmil/examples/plugins/date"
 	"github.com/spf13/cobra"
 )
 
-func main() {
-	cmd := &cobra.Command{
+// CONSTRAINT: All fields of config struct need to be exportable
+type config struct {
+	Key1 string
+	Key2 string
+	Key3 string
+	Key4 string
+}
+
+const cfgFilePath = "./examples/host/config.json"
+
+var (
+	// Stores an instance of the charmil config handler
+	h *c.Handler
+
+	cfg = &config{}
+
+	// Root command of the host CLI
+	cmd = &cobra.Command{
 		Use:          "Host",
 		Short:        "Host CLI for embedding commands",
 		SilenceUsage: true,
 	}
+)
+
+func init() {
+	// Assigns a new instance of the charmil config handler
+	// Links the handler instance to a local config file
+	h = c.New(cfgFilePath, cfg)
+
+	// Loads config values from the local config file
+	err := h.Load()
+	if err != nil {
+		log.Fatal(err)
+	}
+}
+
+func main() {
+	// TODO: Initialize the factory struct here and pass it to the plugins
```

Comment:
So is the idea the plugin will be able to load saved data from the config file when this gets set up.

Reply:
Yes, you're absolutely right.

Just that we won't need to pass the `Handler` as a whole into the factory. Instead, only including the local config file path in the factory and passing it to the plugins will suffice. WDYT?

Reply:
Yeah I think your right, just storing the path would make more sense to me.

Reply:
@jackdelahunt BTW just to update you, according to the latest implementation, we will be including the entire `Handler` in the factory (just as you had mentioned earlier).

Will create an issue/PR for the same shortly.